### PR TITLE
Shared folder feature to  enable or disable shares

### PR DESCRIPTION
**Pull request by stephdl**
A new feature request to enable and disable shares

Screenshot - 2021-03-25T185020 012
Screenshot - 2021-03-25T184958 967
Screenshot (100)
Screenshot (99)

https://github.com/NethServer/dev/issues/6468

### CODE REVIEW

**Review by gsanchietti — COMMENTED** [review]
It looks good to me, but can you please update also the README file and the cockpit API (https://github.com/NethServer/nethserver-cockpit/blob/master/docs/docs/api/nethserver-samba/sharedfolders.md)?

Also I would like a double check from @edospadoni for the UI part and from @DavidePrincipi for the template part.

**Comment by edospadoni** [comment]
It looks good, but in case the shared folder is disabled, the "Enable" button should only show once. But from the screenshot it seems to be duplicated in the three-dot menu as well.

From Firewall application:
Screenshot from 2021-03-26 12-07-30

**Comment by stephdl** [comment]
@edospadoni @gsanchietti In fact when I disable the share, I hide all other options, but I can see in the firewall UI that even if it is disabled you can edit the share

Do you think it is valuable that even in case of disable share  we can still edit and modify acl ?

**Comment by edospadoni** [comment]
> @edospadoni @gsanchietti In fact when I disable the share, I hide all other options, but I can see in the firewall UI that even if it is disabled you can edit the share
> 
> Do you think it is valuable that even in case of disable share we can still edit and modify acl ?

Probably yes, because editing is different from the state (enabled or disabled). I think in the Firewall we have decided to act as you described to allow users to modify the rules, port forward etc ... and then in the future enable or disable them.

File: api/sharedfolders/update [hunk 1]
```diff
@@ -44,6 +44,14 @@ if($input->{'action'} eq 'edit') {
     $eventName = 'ibay-reset-permissions';
 } elsif($input->{'action'} eq 'migrate') {
     $eventName = 'vhost-migrate-ibay';
+} elsif ($input->{'action'} eq 'toggle-lock') {
```
Comment (DavidePrincipi) [inline api/sharedfolders/update:47]: No need to duplicate the "toggle" logic in the API. The API needs just a "set-status" action with one boolean argument or just reuse the existing "edit" one.

**Review by DavidePrincipi — DISMISSED** [review]
The template is ok, however the API design is not convincing.

File: api/sharedfolders/update (hunk 1 above)
Comment (stephdl) [reply api/sharedfolders/update:47]: We use it also in the user account and in the virtualhosts UI, do you think it is a blocking point @DavidePrincipi

Comment (DavidePrincipi) [reply api/sharedfolders/update:47]: Nevermind, I hope you understand my point though

Comment (gsanchietti) [reply api/sharedfolders/update:47]: I like Setph the approach to separate the API with another action: we used the same pattern for all other APIs, one action, one job :)

File: ui/src/views/SharedFolders.vue [hunk 2]
```diff
@@ -208,4 +209,4 @@ export default {
     }
   }
 };
-</script>
\ No newline at end of file
+</script>
```
Comment (gsanchietti) [inline ui/src/views/SharedFolders.vue:212]: Please avoid this change :)

**Review by gsanchietti — APPROVED** [review]
Ok, but remove the extra space from vue file :)

**Review by edospadoni — APPROVED** [review]

File: api/sharedfolders/update (hunk 1 above)
Comment (DavidePrincipi) [reply api/sharedfolders/update:47]: Indeed that's not the point. "Toggle" is a UI behavior and it makes no sense to implement it at the API level too. 
- The API code reads "status" from the DB to decide what to do. 
- The API returns "status" to the UI to let it decide what to show.

IMO the UI should pass the new "status" value to the API and this one has to apply that value.  The line R49 is useless.

I hope you see the point.

**Comment by stephdl** [comment]
I removed some cosmetic lines

in api/sharedfolders/update
in /home/stephdl/dev/git_work/nethserver-samba/ui/src/views/SharedFolders.vue

**Pull request merged**